src/enforce/runners/none_of/index.js

```diff
@@ -3,23 +3,11 @@ import run from '../run';
 
 export default function noneOf(value: mixed, tests: Tests, rules: Rules): boolean {
 
-    const testsCount: number = Object.keys(tests).length;
-    let failCount: number = 0;
+    const validations: Array<string> = Object.keys(tests);
 
-    if (testsCount === 0) {
+    if (validations.length === 0) {
         return false;
     }
 
-    for (const key: string in tests) {
-
-        const success: boolean = run(value, key, tests, rules);
-
-        if (success === true) {
-            continue;
-        } else {
-            failCount++;
-        }
-    }
-
-    return failCount === testsCount;
+    return validations.every((key) => run(value, key, tests, rules) !== true);
```

Comment:
I think we can squeeze another performance boost from this one by:
```js
!validations.some((key) => run(value, key, tests, rules) === true);
```

Since it only takes one positive result to fail the whole set of tests, there is no need to carry on after a positive result is found.

Reply:
That is not necessary. The [Array.prototype.every](http://www.ecma-international.org/ecma-262/6.0/#sec-array.prototype.every) algorithm already short circuits on a falsey result.